search-commons/src/main/java/no/unit/nva/search2/ResourceAwsClient.java

```diff
@@ -96,47 +106,46 @@ private QueryBuilderSource searchSourceWithAggregation(QueryBuilder queryBuilder
 
         builder.size(query.getValue(SIZE).as());
         builder.from(query.getValue(FROM).as());
-        Arrays.stream(query.getValue(SORT).<String>as().split(COMMA))
-            .map(sort -> sort.split(COLON))
-            .map(this::expandSortKeys)
-            .forEach(params -> builder.sort(params.v1(), params.v2()));
+        getSortStream(query).forEach(orderTuple -> builder.sort(orderTuple.v1(), orderTuple.v2()));
 
-        return new QueryBuilderSource(builder, query.openSearchUri(), queryBuilder.mediaType());
-    }
-
-    private Tuple<String, SortOrder> expandSortKeys(String... strings) {
-        var sortOrder = strings.length == 2 ? SortOrder.fromString(strings[1]) : SortOrder.ASC;
-        var luceneKey = SortKeys.keyFromString(strings[0]).getLuceneField();
-        return new Tuple<>(luceneKey, sortOrder);
+        return new QueryBuilderSourceWrapper(builder, query.openSearchUri(), query.getMediaType());
     }
 
     @JacocoGenerated
-    private HttpRequest httpRequest(QueryBuilderSource qbs) {
-        logger.info(qbs.requestUri().toString());
+    private HttpRequest createRequest(QueryBuilderSourceWrapper qbs) {
+        logger.info(qbs.source().query().toString());
         return HttpRequest
                    .newBuilder(qbs.requestUri())
-                   .headers(ACCEPT, qbs.mediaType(), AUTHORIZATION_HEADER, jwtProvider.getValue().getToken())
+                   .headers(
+                       ACCEPT, qbs.mediaType().toString(),
+                       AUTHORIZATION_HEADER, jwtProvider.getValue().getToken())
                    .POST(HttpRequest.BodyPublishers.ofString(qbs.source().toString())).build();
     }
 
-    private OpenSearchSwsResponse sendHttpRequest(HttpRequest httpRequest) {
-        return
-            attempt(() -> httpClient.send(httpRequest, bodyHandler))
-                .map(this::handleResponse)
-                .orElseThrow();
+    private HttpResponse<String> fetch(HttpRequest httpRequest) {
+        return attempt(() -> httpClient.send(httpRequest, bodyHandler)).orElseThrow();
     }
 
     @JacocoGenerated
-    private OpenSearchSwsResponse handleResponse(HttpResponse<String> response) throws BadGatewayException {
-        if (response.statusCode() != HttpStatus.SC_OK) {
-            logger.error(response.body());
-            throw new BadGatewayException(response.body());
+    private OpenSearchSwsResponse handleResponse(HttpResponse<String> response) {
+        if (response.statusCode() != HTTP_OK) {
+            throw new RuntimeException(response.body());
```

Comment:
returning raw error message to client?

Reply:
only validation errors will be returned...